.github/workflows/main.yml

```
@@ -0,0 +1,48 @@
+name: gpt-review
+
+on:
+  pull_request:
+    types: ["opened", "synchronize"]
+    branches:
+      - main
+
+jobs:
+  gpt-review:
+    name: gpt-review
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    env:
+      GH_TOKEN: ${{ github.token }}
+    steps:
+      - name: Checks-out my repository
+        uses: actions/checkout@v4
+      - name: get diff
+        run: gh pr diff ${{ github.event.pull_request.number }} --repo ${{ github.repository }} --color never > diff.txt
+      - name: python set up
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+      - name: install library
+        run: |
+          python -m venv .venv
+          source .venv/bin/activate
+          pip install --upgrade pip
+          pip install -U openai python-dotenv
+          deactivate
+      - name: review from gpt
+        env:
+          AZURE_OPENAI_API_KEY: ${{ secrets.AZURE_OPENAI_API_KEY }}
+          AZURE_API_BASE: ${{ secrets.AZURE_API_BASE }}
+          AZURE_API_VERSION: ${{ secrets.AZURE_API_VERSION }}
+          AZURE_DEPLOY_MODEL: ${{ secrets.AZURE_DEPLOY_MODEL }}
+        run: |
+          touch tmp.txt
+          source .venv/bin/activate
+          python main.py
```

*パフォーマンス* 観点の *medium* レベルの指摘
 ここでは毎回venvを作成し、pipを更新し、必要なライブラリをインストールしています。これには時間がかかる可能性があります。

```suggestion
venvとライブラリのインストールは、必要な時だけ実行するようにするか、またはDockerイメージを使用して依存関係のインストールを事前に行い、実行時にはそのイメージを使用するように変更してみてください。
```